src/sd-boot.c

```diff
@@ -324,6 +322,22 @@ sdb_get_entry_list(sdb_entry_list_t *result)
 	if (token_id == NULL && exists_efi_dir(machine_id))
 		token_id = machine_id;
 
+	return token_id;
+}
+
+bool
+sdb_get_entry_list(sdb_entry_list_t *result)
+{
+	const char *token_id = NULL;
+	DIR *d = NULL;
+	struct dirent *dir;
+	char *path = "/boot/efi/loader/entries";
+
+	memset(result, 0, sizeof(*result));
```

Comment:
It's just a matter of taste, but I usually like the caller to initialize a variable before it gets passed into some function like sdb_get_entry_list.

The slightly bigger issue is the risk of stack buffer overruns if the file has more than SD_MAX_ENTRIES matching entries. Somewhere inside the while (readdir()) loop you should check num_entries and error out when it exceeds the limit.

Reply:
> caller to initialize a variable
Done.

> bigger issue is the risk of stack buffer overruns
Should be done